spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -1226,12 +1227,51 @@ private void rollback(long deliveryTag, Exception e) {
 			}
 		}
 
-		protected synchronized void sendAck(long now) throws IOException {
-			getChannel().basicAck(this.latestDeferredDeliveryTag, true);
+		protected synchronized void sendAck(long now) throws Exception {
+			sendAckWithNotify(this.latestDeferredDeliveryTag, true);
 			this.lastAck = now;
 			this.pendingAcks = 0;
 		}
 
+		/**
+		 * Send ack and notify MessageAckListener(if set).
+		 * @param deliveryTag DeliveryTag of this ack.
+		 * @param multiple Whether multiple ack.
+		 * @throws Exception Occured when ack.
+		 */
+		private void sendAckWithNotify(long deliveryTag, boolean multiple) throws Exception {
+			try {
+				getChannel().basicAck(deliveryTag, multiple);
+				notifyMessageAckListener(getMessageAckListener(), true, deliveryTag, null);
+			}
+			catch (Exception e) {
```

Comment:
This can be `IOException` since `notifyMessageAckListener` does not throw any exception - we can then remove all those `throws ...` and `catch (Exception...` changes.

Reply:
The reason why it was changed to ```Exception``` at that time was because ```channel#ack``` might throw other types of exceptions (not ```IOexception```) during the test.